app/models/tag.rb

```diff
@@ -1023,7 +1033,7 @@ def remove_sub_filters(sub_tag)
   before_update :check_canonical
   def check_canonical
     if self.canonical_changed? && !self.canonical?
-      self.async(:remove_canonical_associations)
+      remove_canonical_associations
```

Comment:
They're synchronous, but they don't act on the same object in memory. Changes in one won't necessarily be reflected in the other. When setting `syn_string`, for instance, it updates the value of the merger on the object that you're modifying. But when `add_merger_associations` is called asynchronously in the `syn_string=` function, the new merger hasn't yet been saved to the database, so it'll load the version of the tag with no merger (and it won't bother transferring any associations).

I may be a little overzealous -- it might be okay to call `remove_canonical_associations` asynchronously (although it seems less computationally intensive than `add_merger_associations`, which does require special handling). But it is in a `before_update` callback, which makes me nervous.